src/System.Linq/tests/OrderByTests.cs

```diff
@@ -422,5 +422,19 @@ public void SortsRandomizedEnumerableCorrectly(int items)
             Array.Sort(randomized);
             Assert.Equal(randomized, ordered);
         }
+
+        [Theory]
+        [InlineData(new[] { 1 })]
+        [InlineData(new[] { 1, 2 })]
+        [InlineData(new[] { 2, 1 })]
+        [InlineData(new[] { 1, 2, 3, 4, 5 })]
+        [InlineData(new[] { 5, 4, 3, 2, 1 })]
+        [InlineData(new[] { 4, 3, 2, 1, 5, 9, 8, 7, 6 })]
+        [InlineData(new[] { 2, 4, 6, 8, 10, 5, 3, 7, 1, 9 })]
+        public void TakeOne(IEnumerable<int> source)
+        {
+            foreach (int x in source.OrderBy(i => i).Take(1))
+                Assert.Equal(source.Min(), x);
```

Comment:
Are there tests elsewhere that validate OrderBy(...).Take(1) returns 1 and only 1 item if there's at least one item in the input?  Since we're modifying those code paths, it'd be good to validate that here, too, e.g.
```C#
int count = 0;
foreach (int x in source.OrderBy(i => i).Take(1))
{
    Assert.Equal(source.Min(), x);
}
Assert.Equal(1, count);
```

Reply:
Fixed.